src/main/java/com/github/maxopoly/finale/combat/AsyncPacketHandler.java

```diff
@@ -0,0 +1,76 @@
+package com.github.maxopoly.finale.combat;
+
+import org.bukkit.World;
+import org.bukkit.entity.Damageable;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Player;
+
+import com.comphenix.protocol.PacketType;
+import com.comphenix.protocol.events.ListenerPriority;
+import com.comphenix.protocol.events.PacketAdapter;
+import com.comphenix.protocol.events.PacketContainer;
+import com.comphenix.protocol.events.PacketEvent;
+import com.comphenix.protocol.reflect.StructureModifier;
+import com.comphenix.protocol.wrappers.EnumWrappers.EntityUseAction;
+import com.github.maxopoly.finale.Finale;
+
+import net.md_5.bungee.api.ChatColor;
+
+public class AsyncPacketHandler extends PacketAdapter {
```

Comment:
So does this replace the vanilla handler or does it act on top of it?

Reply:
Replaces.

Reply:
https://en.wikipedia.org/wiki/Producer%E2%80%93consumer_problem

Actually thinking about it more that's not needed here though, because the consumption is sync anyway. So never mind.